packages/vhd-lib/Vhd/VhdDirectory.js

```diff
@@ -131,6 +132,7 @@ exports.VhdDirectory = class VhdDirectory extends VhdAbstract {
     this._path = path
     this._opts = opts
     this.#compressor = getCompressor(opts?.compression)
+    this.writeBlockAllocationTable = synchronized.withKey()(this.writeBlockAllocationTable)
```

Comment:
No need to use `withKey` because this method is called without arguments:

```suggestion
    this.writeBlockAllocationTable = synchronized()(this.writeBlockAllocationTable)
```